device.yml

```diff
@@ -0,0 +1,330 @@
+  DO3Threshold:
+        <<: *thresholdvalue
+        address: 69
+        description: Value used to threshold an ADC read, and trigger DO3 pin.
+  Reserved17:
+      <<: *ReservedReg
+      address: 70
+  Reserved18:
+      <<: *ReservedReg
+      address: 71
+  Reserved19:
+      <<: *ReservedReg
+      address: 72
+  Reserved20:
+      <<: *ReservedReg
+      address: 73
+  DO0BufferRisingEdge: &bufferedthreshold
+      address: 74
+      access: Write
+      type: U16
+      defaultValue: 0
+      description: Time (ms) above threshold value that is required to trigger a DO0 pin event.
+  DO1BufferRisingEdge:
+      <<: *bufferedthreshold
+      address: 75
+      description: Time (ms) above threshold value that is required to trigger a DO1 pin event.
+  DO2BufferRisingEdge:
+      <<: *bufferedthreshold
+      address: 76
+      description: Time (ms) above threshold value that is required to trigger a DO2 pin event.
+  DO3BufferRisingEdge:
+      <<: *bufferedthreshold
+      address: 77
+      description: Time (ms) above threshold value that is required to trigger a DO3 pin event.
+  Reserved21:
+      <<: *ReservedReg
+      address: 78
+  Reserved22:
+      <<: *ReservedReg
+      address: 79
+  Reserved23:
+      <<: *ReservedReg
+      address: 80
+  Reserved24:
+      <<: *ReservedReg
+      address: 81
+  DO0BufferFallingEdge:
+      <<: *bufferedthreshold
+      address: 82
+      description: Time (ms) below threshold value that is required to trigger a DO0 pin event.
+  DO1BufferFallingEdge:
+      <<: *bufferedthreshold
+      address: 83
+      description: Time (ms) below threshold value that is required to trigger a DO1 pin event.
+  DO2BufferFallingEdge:
+      <<: *bufferedthreshold
+      address: 84
+      description: Time (ms) below threshold value that is required to trigger a DO2 pin event.
+  DO3BufferFallingEdge:
+      <<: *bufferedthreshold
+      address: 85
+      description: Time (ms) below threshold value that is required to trigger a DO3 pin event.
+  Reserved25:
+      <<: *ReservedReg
+      address: 86
+  Reserved26:
+      <<: *ReservedReg
+      address: 87
+  Reserved27:
+      <<: *ReservedReg
+      address: 88
+  Reserved28:
+      <<: *ReservedReg
+      address: 89
+  Reserved29:
+      <<: *ReservedReg
+      address: 90
+bitMasks:
+  DigitalOutputs:
+    description: Specifies the state of port digital output lines.
+    bits:
+      DO0: 0x1
+      DO1: 0x2
+      DO2: 0x4
+      DO3: 0x8
+      DO0Changed: 0x10
+      DO1Changed: 0x20
+      DO2Changed: 0x40
+      DO4Changed: 0x80
+groupMasks:
+  EnableFlag:
+    description: The state of an abstract functionality.
+    values:
+      Disabled: 0
+      Enabled: 1
+  DigitalInputState:
+    description: The state of a digital input pin.
+    values:
+      Low: 0
+      High: 1
+  RangeAndBandwidthConfig:
+    description: The available settings for set the range and bandwidth of the ADC.
+    values:
+      Range5V_Bandwidth1k5: 0x06
```

Comment:
What is bandwidth? Is it some sort of filter? If so I don't understand these values since they frequency of acquisition is 1khz, most of these are wayyyy above that

Reply:
The bandwidth is available on the ADC itself. The options exist because they also exist in the hardware (ADC).
Nevertheless, the hardware allows increasing the sample rate significantly, so the bandwidth options can make sense someday 🤷‍♂️

Reply:
Oki, I will leave it as is. It is a LowPass filter correct?